src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -920,9 +941,15 @@ abstract class ManagedStrategy(
         // emulating spurious wake-ups.
         newSwitchPoint(iThread, codeLocation, tracePoint)
         parkingTracker.park(iThread)
-        while (parkingTracker.waitUnpark(iThread)) {
+        // forbid spurious wake-ups if inside silent sections
+        var allowSpuriousWakeUp = shouldAllowSpuriousUnpark(iThread, codeLocation)
+        while (parkingTracker.waitUnpark(iThread, allowSpuriousWakeUp)) {
             // switch to another thread and wait till an unpark event happens
-            switchCurrentThread(iThread, BlockingReason.Parked)
+            val switchHappened = switchCurrentThread(iThread, BlockingReason.Parked)
+            // if the thread cannot switch, then allow spurious wake-up
```

Comment:
Why should Lincheck allow spurious wake-ups here? Should the execution finish with a deadlock instead?

Reply:
There is a case I mentioned above.

> Suppose a thread is parked inside a silent section, and all other threads are blocked.
> Shouldn't the strategy then attempt to unpark it spuriously?

If we don't want to unpark in this case, then perhaps it is safe to revert this.

Reply:
Reverted